scylla/src/transport/cql_types_test.rs

```diff
@@ -1149,6 +1149,72 @@ async fn test_timeuuid() {
     }
 }
 
+#[tokio::test]
+async fn test_timeuuid_ordering() {
+    let session: Session = create_new_session_builder().build().await.unwrap();
+    let ks = unique_keyspace_name();
+
+    session
+        .query(
+            format!(
+                "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = \
+            {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1}}",
+                ks
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
+    session.use_keyspace(ks, false).await.unwrap();
+
+    session
+        .query(
+            "CREATE TABLE tab (p int, t timeuuid, PRIMARY KEY (p, t))",
+            (),
+        )
+        .await
+        .unwrap();
+
+    // Timeuuid values, sorted in the same order as Scylla/Cassandra sorts them.
+    let sorted_timeuuid_vals: Vec<CqlTimeuuid> = vec![
+        CqlTimeuuid::from_str("00000000-0000-1000-8080-808080808080").unwrap(),
+        CqlTimeuuid::from_str("00000000-0000-1000-ffff-ffffffffffff").unwrap(),
+        CqlTimeuuid::from_str("00000000-0000-1000-0000-000000000000").unwrap(),
+        CqlTimeuuid::from_str("fed35080-0efb-11ee-a1ca-00006490e9a4").unwrap(),
+        CqlTimeuuid::from_str("00000257-0efc-11ee-9547-00006490e9a6").unwrap(),
+        CqlTimeuuid::from_str("ffffffff-ffff-1fff-ffff-ffffffffffef").unwrap(),
+        CqlTimeuuid::from_str("ffffffff-ffff-1fff-ffff-ffffffffffff").unwrap(),
+        CqlTimeuuid::from_str("ffffffff-ffff-1fff-0000-000000000000").unwrap(),
+        CqlTimeuuid::from_str("ffffffff-ffff-1fff-7f7f-7f7f7f7f7f7f").unwrap(),
+    ];
```

Comment:
Again, if we test sorting, it would be good to have unsorted data. You can use https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.shuffle to avoid having 2 literals in the code.

Reply:
We should also use a random generator with known algorithm - `std::thread_rng` uses an unspecified algorithm and can probably change in the future. We already have tests with fixed pseudorandomness - for example, see `partitioners_output_same_result_no_matter_how_input_is_partitioned`.

Reply:
Also, `362880` is not a huge number. Perhaps checking all the permutations would not take much time (let's aim under 100ms)? We can iterate over permutations using [`Itertools::permutations()`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.permutations).